src/main/java/com/entropyteam/entropay/employees/services/ContractService.java

```diff
@@ -28,14 +33,45 @@ public class ContractService extends BaseService<Contract, ContractDto, UUID> {
 
     @Autowired
     public ContractService(ContractRepository contractRepository, CompanyRepository companyRepository,
-            EmployeeRepository employeeRepository, RoleRepository roleRepository,
-            SeniorityRepository seniorityRepository) {
+                           EmployeeRepository employeeRepository, RoleRepository roleRepository,
+                           SeniorityRepository seniorityRepository) {
         this.contractRepository = contractRepository;
         this.companyRepository = companyRepository;
         this.employeeRepository = employeeRepository;
         this.roleRepository = roleRepository;
         this.seniorityRepository = seniorityRepository;
+    }
+
+    @Transactional
+    @Override
+    public ContractDto create(ContractDto contractDto) {
+        contractRepository.findContractByEmployeeAndActiveIsTrue(
+                        employeeRepository.findById(contractDto.employeeId()).orElseThrow()
+                )
+                .ifPresent(existent -> {
+                    existent.setActive(false);
+                    contractRepository.saveAndFlush(existent);
+                });
+        return super.create(contractDto);
```

Comment:
At this point we need to validate that the new contract will be crated with active = true

Reply:
According to the acceptance criteria, all new contracts must be set to active. I was in doubt about how to proceed here since the DTO is a record class and can't be modified.

Reply:
One approach could be add a validation like:
```
if (!contractDto.active) {
  return => validation error
}

```
Other is, instead of call `return super.create(contractDto);` rewrite the save logic, like:

```
Contract contract = toEntity(entity);
contract.setActive(true);
Entity savedEntity = getRepository().save(contract);
return toDTO(savedEntity);
```

Reply:
I'm not a fan of throwing an error if the creation of a new contract implies always an active contract.
Regarding the second approach, we'd be converting the same object two times. Could this be avoided? Is it necessary for the DTOs to be a record class?

Reply:
I don't see where we're converting the same object twice, but yes i think we can avoid that
Yes for the leverage Record classes

Reply:
I'm using Lombok @With annotation which simplifies this a lot.